tools/cli/commands.js

```diff
@@ -271,11 +271,30 @@ function isModernArchsOnlyEnabled(appDir) {
   return packageJson?.meteor?.modernWebArchsOnly === true;
 }
 
-function filterWebArchs(webArchs, excludeArchsOption, appDir) {
-  const automaticallyIgnoredLegacyArchs = (appDir && isModernArchsOnlyEnabled(appDir)) ? ['web.browser.legacy', 'web.cordova'] : [];
-  if (excludeArchsOption || automaticallyIgnoredLegacyArchs.length) {
-    const excludeArchs = [...(excludeArchsOption ? excludeArchsOption.trim().split(/\s*,\s*/) : []), ...automaticallyIgnoredLegacyArchs];
-    webArchs = webArchs.filter(arch => !excludeArchs.includes(arch));
+function filterWebArchs(webArchs, excludeArchsOption, appDir, options) {
+  const platforms = (options.platforms || []);
+  const isBuildMode = platforms?.length > 0;
+  if (isBuildMode) {
+    // Build Mode
+    const isModernOnlyPlatform = platforms.includes('modern') && !platforms.includes('legacy');
+    if (isModernOnlyPlatform) {
+      webArchs = webArchs.filter(arch => arch !== 'web.browser.legacy');
+    }
+    const hasCordovaPlatforms = platforms.includes('android') || platforms.includes('ios');
+    if (!hasCordovaPlatforms) {
+      webArchs = webArchs.filter(arch => arch !== 'web.cordova');
+    }
```

Comment:
Now, if you add `modern` within `.meteor/local` you establish modern only build. Otherwise, just having `browser` will include both, legacy and modern archs when `meteor build`.